soc/silabs/silabs_siwx91x/siwg917/siwx91x_nwp.c

```diff
@@ -147,13 +152,53 @@ static void siwx91x_apply_sram_config(sl_si91x_boot_configuration_t *boot_config
 	}
 }
 
+static void siwx91x_apply_nwp_config(const struct device *dev,
+				     sl_si91x_boot_configuration_t *boot_config)
+{
+	const struct siwx91x_nwp_config *cfg = dev->config;
+	struct {
+		bool enabled;
+		uint32_t *target;
+		uint32_t bit;
+	} features[] = {
+		{ cfg->rtc_from_host, &boot_config->custom_feature_bit_map,
+		  SL_SI91X_CUSTOM_FEAT_RTC_FROM_HOST },
+		{ cfg->support_1p8v, &boot_config->ext_custom_feature_bit_map,
+		  SL_SI91X_EXT_FEAT_1P8V_SUPPORT },
+		{ cfg->disable_xtal_correction, &boot_config->ext_custom_feature_bit_map,
+		  SL_SI91X_EXT_FEAT_DISABLE_XTAL_CORRECTION },
+		{ cfg->qspi_80mhz_clk, &boot_config->ext_custom_feature_bit_map,
+		  SL_SI91X_EXT_FEAT_NWP_QSPI_80MHZ_CLK_ENABLE },
+	};
+
+	for (int i = 0; i < ARRAY_SIZE(features); i++) {
+		if (features[i].enabled) {
+			*features[i].target |= features[i].bit;
+		}
+	}
+
+	if (cfg->antenna_selection) {
```

Comment:
In fact, you can write:

    #define SL_SI91X_EXT_FEAT_FRONT_END_MSK (BIT(30) | BIT(29))
    [...]

    boot_config->ext_custom_feature_bit_map &= ~SL_SI91X_EXT_FEAT_FRONT_END_MASK;
    boot_config->ext_custom_feature_bit_map |= FIELD_PREP(SL_SI91X_EXT_FEAT_FRONT_END_MSK,
                                                          cfg->antenna_selection);

    [...]
    .antenna_selection = DT_INST_ENUM_IDX(inst, antenna_selection),